haskell/src/Monocle/Class.hs

```diff
@@ -24,6 +24,13 @@ instance MonadTime IO where
   mGetCurrentTime = Data.Time.Clock.getCurrentTime
   mThreadDelay = Control.Concurrent.threadDelay
 
+-------------------------------------------------------------------------------
+-- A concurrent system handled via Control.Concurrent.MVar (Unlifted from IO)
+
+class MonadUnliftIO m => MonadSync m where
```

Comment:
By having the constraint here makes IO available to any MonadSync user. I don't think it matter for this particular class, thus I would recommend to remove ths constraint.

And we should provide an instance for IO, so that we can keep the IO instance of MonadGraphQL below:

```haskell
class MonadSync IO where
  mNewMVar = newMVar
  mModifyMVar = modifyMVar
```